jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java

```diff
@@ -31,8 +31,7 @@ public void saveToImage(BufferedImage bufferedImage) {
     public abstract void apply(int x, int y);
 
     protected void setPixel(int x, int y, Pixel pixel) {
-        int[] sourceColors = new int[4];
-        source.getPixel(x, y, sourceColors);
+        int[] sourceColors = source.getPixel(x, y, (int[]) null);
```

Comment:
we should just use our method getPixel(x, y).getColors() in case we change out source implementation in the future